src/VisualStudio/Core/Def/LanguageService/AbstractPackage`2.cs

```diff
@@ -36,20 +37,10 @@ protected AbstractPackage()
 
     protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
     {
-        await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
-
-        await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-
-        var shell = (IVsShell7)await GetServiceAsync(typeof(SVsShell)).ConfigureAwait(true);
-        var solution = (IVsSolution)await GetServiceAsync(typeof(SVsSolution)).ConfigureAwait(true);
-        cancellationToken.ThrowIfCancellationRequested();
-        Assumes.Present(shell);
-        Assumes.Present(solution);
+        // Should only be called from a threadpool. Opinionated, as package load sequence thread switches are impactful.
+        Contract.ThrowIfTrue(JoinableTaskFactory.Context.IsOnMainThread);
```

Comment:
Ditto.

Reply:
I don't think this can be removed, and is safe to do as the derived classes make sure this assumption is valid.